dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+import os
+from collections import OrderedDict
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from dbt.contracts.graph.nodes import Group, ManifestNode
+from dbt.contracts.results import CatalogTable
+from dbt.node_types import AccessType, NodeType
+
+from dbt_meshify.storage.file_manager import DbtFileManager
+
+
+def filter_empty_dict_items(dict_to_filter: Dict[str, Any]):
+    """Filters out empty dictionary items"""
+    return {k: v for k, v in dict_to_filter.items() if v}
+
+
+def process_model_yml(model_yml: Dict[str, Any]):
+    """Processes the yml contents to be written back to a file"""
+    model_ordered_dict = OrderedDict.fromkeys(
+        [
+            "name",
+            "description",
+            "latest_version",
+            "access",
+            "group",
+            "config",
+            "meta",
+            "tests",
+            "columns",
+            "versions",
+        ]
+    )
+    model_ordered_dict.update(model_yml)
+    # remove any keys with None values
+    return filter_empty_dict_items(model_ordered_dict)
+
+
+def resources_yml_to_dict(resources_yml: Dict, resource_type: NodeType = NodeType.Model):
+    """Converts a yml dict to a named dictionary for easier operation"""
+    return (
+        {resource["name"]: resource for resource in resources_yml[resource_type.pluralize()]}
+        if resources_yml
```

Comment:
Double check the type hints for `resource_yml`. If this variable can be false, then it's type cannot just be `Dict`. Maybe `Optional[Dict]`?